qiita_pet/handlers/auth_handlers.py

```diff
@@ -175,3 +176,61 @@ class AuthLogoutHandler(BaseHandler):
     def get(self):
         self.clear_cookie("user")
         self.redirect("%s/" % qiita_config.portal_dir)
+
+
+class GlobusOAuth2LoginHandler(BaseHandler, GlobusOAuth2Mixin):
+    # Redirect to Globus Auth
+    def post(self):
+        self.authorize_redirect(
+            redirect_uri=qiita_config.globus_redirect_uri,
+            client_id=qiita_config.globus_client_key,
+            scope=["openid", "profile", "email", "urn:globus:auth:scope:transfer.api.globus.org:all"],
+            response_type="code",
+            extra_params={"access_type": "offline"})
+
+    # Process a redirect from Globus Auth, exchange the code to
+    # access/refresh tokens,and get userinfo
+    async def get(self):
+        if self.get_argument("code", False):
+            nextpage = self.get_argument("next", None)
+            if nextpage is None:
+                if "auth/" not in self.request.headers["Referer"]:
+                    nextpage = self.request.headers["Referer"]
+                else:
+                    nextpage = "%s/" % qiita_config.portal.portal_dir
+            # Exchange the code to access/refresh tokens
+            tokens = await self.get_tokens(
+                key=qiita_config.globus_client_key,
+                secret=qiita_config.globus_client_secret,
+                redirect_uri=qiita_config.globus_redirect_uri,
+                code=self.get_argument("code"))
+            # Get userinfo from Globus Auth
+            user_info = await self.get_user_info(tokens["access_token"])
+
+            # Check if the user exists. If it does not, create the user
+            username = user_info.get("preferred_username")
```

Comment:
Are you sure is fine to leave this info here? Also, when we create a new user, will this be in the Qiita system or Globus? I think it should be in the latter, right?

Reply:
I removed the comment. Users do not create Qiita accounts. A new local Qiita account is created automatically, when a user logs in to Qiita via Globus Auth for the first time. A local Qiita account is a tornado/Qiita version of a copy of a Globus account.